### DISCUSSION TITLE
Theme flickers when setting theme

### DISCUSSION THREAD

**Discussion by YathindraNikshipth** [eclipse-theia/theia#14656] [General]
I'm trying to set the Theia theme using the setCurrentTheme method of the ThemeService before the loader starts. However, multiple checks repeatedly call setCurrentTheme, causing a theme flicker until Theia is fully initialized.

I believe I need to rebind ThemeServiceWithDB from @theia/monaco instead of ThemeService and override the setCurrentTheme method to meet my requirements, i binded my custom Class as follows:

`rebind(ThemeServiceWithDB).to(CustomThemeServiceWithDB).inSingletonScope()`

When I rebind ThemeServiceWithDB and override the setCurrentTheme method as required, the theme still flickers. However, when I go to the settings tab in the UI and manually change the color theme, the selected theme is not reflected.

```ts
setCurrentTheme(themeId: string, persist = true): void {

        if (themeId !== 'my_theme') {
            return;
        }

        super.setCurrentTheme(themeId, persist);

    }
```
Could you please let me know where I’m going wrong?

**Comment by msujew** [comment]
Hey @YathindraNikshipth,

what exactly are you trying to accomplish? Do you want Theia to start up with your custom theme in place?

> However, when I go to the settings tab in the UI and manually change the color theme, the selected theme is not reflected.

You mean your custom theme, or any other theme? Because your code leads me to believe that only your custom theme can be selected, and upon selecting any other theme, the method just returns.

**Reply by YathindraNikshipth** [reply]
@msujew yes I'm trying to start theia with my custom theme in place.

[

https://github.com/user-attachments/assets/d74d71b0-3c48-4965-958b-49c5a1767cee

](url)


In the video, you can see that a white theme initially appears before switching to the theme of my choice. I'm trying to figure out where this is happening. I've overridden the FrontendApplication's start method to set the theme, but I'm still experiencing the flicker

**Reply by msujew** [reply]
Which version of Theia are you using? I believe that this is a duplicate of https://github.com/eclipse-theia/theia/issues/14428 which was fixed in version 1.56.0 of Theia, see https://github.com/eclipse-theia/theia/pull/14491.

**Reply by YathindraNikshipth** [reply]
Hello @msujew,

I wanted to check if this issue has been resolved in the browser version as well. I'm still noticing a flicker from dark to white after upgrading to v1.56.0.

Additionally, I have a question: Is there a promise I can await in order to keep the loader active until the host-plugin process is completed?

**Reply by msujew** [reply]
I'll need to take a look at the code. I'll get back to this once I'm back from vacation in the new year.

**Reply by YathindraNikshipth** [reply]
@msujew  Enjoy your vacation! We’ll catch up when you're back in the new year!
